UsageExamples/DateHelperExample.cs

```diff
@@ -0,0 +1,290 @@
+    /// <returns></returns>
+    public static (DateTime earlyOpen, DateTime normalOpen, DateTime normalClose, DateTime lateClose) GetMarketHours(DateTime date)
+    {
+        var defaultDate = new DateTime(date.Year, date.Month, date.Day, 0, 0, 0);
+
+        if (IsTodayAWeekDay(date))
+        {
+            var holidays = GetHolidays(date.Year);
+            var dateSearch = holidays.Where(x => x.Date.Year == date.Year && x.Date.Month == date.Month && x.Date.Day == date.Day).ToList();
+            var result = dateSearch.FirstOrDefault();
+
+            // changed early and late hours to match new alpaca times
+            var earlyOpen = new DateTime(date.Year, date.Month, date.Day, 4, 0, 0);
+            var normalOpen = new DateTime(date.Year, date.Month, date.Day, 9, 30, 0);
+            var normalClose = new DateTime(date.Year, date.Month, date.Day, 16, 0, 0);
+            var lateClose = new DateTime(date.Year, date.Month, date.Day, 20, 0, 0);
+
+            if (dateSearch.Count == 0)
+            {
+                return (earlyOpen, normalOpen, normalClose, lateClose);
+            }
+            else
+            {
+                // only checking for early hours for christmas eve and black friday and sometimes independence day
+                if (result.Hour != 0 && date.Hour < result.Hour)
+                {
+                    var holidayClose = new DateTime(result.Year, result.Month, result.Day, result.Hour, result.Minute, result.Second);
+                    return (earlyOpen, normalOpen, holidayClose, holidayClose);
+                }
+            }
+        }
+
+        return (defaultDate, defaultDate, defaultDate, defaultDate);
+    }
+
+    /// <summary>
+    /// Finds out if today is a week day
+    /// </summary>
+    /// <param name="date"></param>
+    /// <returns></returns>
+    private static bool IsTodayAWeekDay(DateTime date = default)
+    {
+        return date.DayOfWeek != DayOfWeek.Saturday && date.DayOfWeek != DayOfWeek.Sunday;
+    }
+
+    /// <summary>
+    /// Calculates the Easter Sunday date for the selected year
+    /// </summary>
+    /// <param name="year"></param>
+    /// <returns></returns>
+    private static DateTime EasterSunday(int year)
+    {
+        int g = year % 19;
+        int c = year / 100;
+        int h = (c - c / 4 - ((8 * c + 13) / 25) + 19 * g + 15) % 30;
+        int i = h - h / 28 * (1 - h / 28 * (29 / (h + 1)) * ((21 - g) / 11));
+
+        int day = i - (year + year / 4 + i + 2 - c + c / 4) % 7 + 28;
+        int month = 3;
+        if (day > 31)
+        {
+            month++;
+            day -= 31;
+        }
+
+        return new DateTime(year, month, day, 0, 0, 0);
+    }
+
+    /// <summary>
+    /// Returns a list of all market holiday dates with special closed hours or shortened market hours for selected year
+    /// </summary>
+    /// <param name="year"></param>
+    /// <returns></returns>
+    private static HashSet<DateTime> GetHolidays(int year)
```

Comment:
This method is too big. Extract the helper methods for each holiday as you've already done for the Easter calculaiton.